apps/aurora-portal/src/server/Network/routers/networkrouter.ts

```diff
@@ -0,0 +1,16 @@
+import { protectedProcedure } from "../../trpc"
+import { networkResponseSchema, Network } from "../types/models"
+
+export const networkRouter = {
+  getNetworks: protectedProcedure.query(async ({ ctx }): Promise<Network[] | undefined> => {
+    const openstackSession = ctx.openstack
```

Comment:
Please rename the function to getRoutersByProjectId, add input with zod validation. And please call `ctx.rescopeSession({projectId: input.projectId})`instead of `ctx.openstack`.

```ts
.input(z.object({projectId:z.string()}))
```